lib/units/validateBody.js

```diff
@@ -197,9 +199,22 @@ function validateBody(expected, real) {
     : getBodyValidator(realType, expectedType);
 
   if (validatorError) {
-    errors.push({
-      message: validatorError
-    });
+    // In case determined media types mismtach, check if the real is not missing.
+    // Keep in mind the following scenarios:
+    // 1. Expcected '', and got '' (TextDiff/TextDiff, valid)
+    // 2. Expected {...}, but got '' (Json/TextDiff, invalid, produces "missing real body" error)
+    // 3. Expected {...}, but git "foo" (Json/TextDiff, invalid, produces types mismatch error).
+    if (expected.body !== '' && hasEmptyRealBody) {
```

Comment:
This isn't the best solution in terms of cyclomatic complexity, but the one that works without introducing an entirely custom flow for missing body. This way we can always coerce missing actual "body" to an empty string and properly differentiate between three mentioned scenarios.